src/components/Button/Button.js

```diff
@@ -0,0 +1,40 @@
+import React, { useContext } from 'react'
+import UsersContext from "../../Context/UsersContext"
+import RequestRoomieContext from '../../Context/RequestRoomieContext'
+
+export default function Button({ id, email }) {
+  const { currentUser } = useContext(UsersContext)
+  const {requestRoomie, deleteRoomie } = useContext(RequestRoomieContext)
+
+  const cursor = {cursor: 'pointer'}
+
+  const changeRequest = (query) => {
+    const reqBody = {
+      "requestor_id": currentUser.id,
+      "receiver_id": id
+    }
+    requestRoomie(reqBody, query)
+  }
+
+  const removeRequest = () => {
+    const remBody = {
+      "requestor_id": id,
+      "receiver_id": currentUser.id
+    }
+    deleteRoomie(remBody)
+  }
+
+  return (
+    <div>
+      {!currentUser ? null :
```

Comment:
This chaining ternary is GENIUS! Love it, this functionality seemed daunting to me yet this was such a clever, sleek answer!